utils/providers.py

```diff
@@ -762,16 +762,17 @@ def get_crud(provider_config_name):
             key=provider_config_name)
     elif prov_type == 'openstack':
         from cfme.cloud.provider import OpenStackProvider
+
+        custom_credentials = {'default': credentials}
         if 'amqp_credentials' in prov_config:
-            amqp_credentials = process_credential_yaml_key(
+            custom_credentials["amqp"] = process_credential_yaml_key(
                 prov_config['amqp_credentials'], cred_type='amqp')
+
         return OpenStackProvider(name=prov_config['name'],
             hostname=prov_config['hostname'],
             ip_address=prov_config['ipaddress'],
             api_port=prov_config['port'],
-            credentials={
-                'default': credentials,
-                'amqp': amqp_credentials},
+            credentials=custom_credentials,
```

Comment:
It probably does not, the problem here was that amqp_credentials was not mandatory and created only if present but referenced every time.